src/tds/config.c

```diff
@@ -466,10 +466,10 @@ tds_config_boolean(const char *option, const char *value, TDSLOGIN *login)
 static void
 tds_config_encryption(const char * value, TDSLOGIN * login)
 {
-	TDS_ENCRYPTION_LEVEL lvl = TDS_ENCRYPTION_OFF;
+	TDS_ENCRYPTION_LEVEL lvl = TDS_ENCRYPTION_REQUEST;
 
 	if (!strcasecmp(value, TDS_STR_ENCRYPTION_OFF))
-		;
+		lvl = TDS_ENCRYPTION_OFF;
```

Comment:
I recall single-stepping through this function. But I think perhaps my changes here are needed, plus I need to change tds_alloc_login also, when there is no encryption configuration.

It is line 469 where I change the default` lvl to TDS_ENCRYPTION_REQUEST `as you suggested for backward compatibility. Then if the user specified `encryption = off `in the freetds.conf file this line of code 472 is needed, whereas in the past it would not be needed, but would certainly have been OK to have written.  With the change in line 469 one could, if one was obsessive about optimizations, remove line of code 474, since this would be a NOOP. However, I choose not to optimize code to the extent that it is confusing so leave this line of code in place: `lvl = TDS_ENCRYPTION_REQUEST`.